src/wp-includes/template.php

```diff
@@ -921,6 +922,21 @@ function wp_start_template_enhancement_output_buffer(): bool {
  * @return string Finalized output buffer.
  */
 function wp_finalize_template_enhancement_output_buffer( string $output, int $phase ): string {
+	$do_send_late_headers = static function ( string $output ): void {
+		/**
+		 * Fires at the last moment HTTP headers may be sent.
+		 *
+		 * This happens immediately before the template enhancement output buffer is flushed. This is in contrast with
+		 * the {@see 'send_headers'} action which fires after the initial headers have been sent before the template
+		 * has begun rendering, and thus does not depend on output buffering.
+		 *
+		 * @since 6.9.0
+		 *
+		 * @param string $output Output buffer.
+		 */
+		do_action( 'wp_send_late_headers', $output );
```

Comment:
This is very cautious but it runs late: servers and plugins can do unexpected things. 
 
```suggestion
		if ( ! headers_sent() ) {
			do_action( 'wp_send_late_headers', $output );
		}
```

Reply:
I'm not sure this is right, because it would silence sending of headers which the site is expecting to be able to send. The developer should have some signal as to why this wasn't done. For a corollary, there is no such `if()` wrapping the `send_headers` action:

https://github.com/WordPress/wordpress-develop/blob/8d24041c08a58b2f79504699fb3f63d01737b876/src/wp-includes/class-wp.php#L578-L595

Reply:
This is _closely_ related to Core-64108. Any notices/warnings/errors should be captured to be properly appended to the output buffer.